src/applications/hca/containers/IDPage.jsx

```diff
@@ -99,15 +104,19 @@ class IDPage extends React.Component {
 }
 
 const mapDispatchToProps = {
+  setData,
   submitIDForm,
   toggleLoginModal,
 };
 
 const mapStateToProps = state => ({
-  shouldHideIDForm: state.hcaIDForm.shouldHideIDForm,
-  showLoadingIndicator: isProfileLoading(state),
+  enrollmentStatus: state.hcaIDForm.enrollmentStatus,
+  errors: state.hcaIDForm.errors,
+  form: state.form,
+  hasOptionalDD214Upload: state.hcaIDForm.hasOptionalDD214Upload,
   isSubmittingIDForm: state.hcaIDForm.isSubmitting,
-  error: state.hcaIDForm.error,
+  shouldRedirect: isLoggedIn(state),
```

Comment:
I realize you probably initially went back and forth on how to handle the logged in state. I think using the reducer to detect the logged in state change is essentially the same as just looking at the state directly (just under a different attribute name), and that just directly accessing the state is still the cleanest way to observe it. That's why I went back to handling the logged in state this way.

Reply:
yeah this is a much better way of handling things than my initial approach.